android-runtime/runtime/src/main/java/io/sdkgen/runtime/SdkgenHttpClient.kt

```diff
@@ -0,0 +1,225 @@
+package io.sdkgen.runtime
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Point
+import android.os.Build
+import android.provider.Settings
+import android.view.WindowManager
+import com.google.gson.*
+import okhttp3.*
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.RequestBody.Companion.toRequestBody
+import java.io.InterruptedIOException
+import java.net.SocketTimeoutException
+import java.util.*
+import java.util.concurrent.TimeUnit
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+
+@Suppress("unused")
+class SdkGenHttpClient(
+    private val baseUrl: String,
+    private val applicationContext: Context,
+    private val defaultTimeoutMillis: Long = 10000L
+) {
+
+    data class InternalResponse(val error: JsonObject?, val result: JsonObject?)
+
+    private val random = Random()
+    private val hexArray = "0123456789abcdef".toCharArray()
+    private val gson = Gson()
+    private val connectionPool = ConnectionPool(10, 5, TimeUnit.MINUTES)
+    private val httpClient = OkHttpClient.Builder()
+        .connectionPool(connectionPool)
+        .dispatcher(Dispatcher().apply { maxRequests = 200 ; maxRequestsPerHost = 200 })
+        .connectTimeout(180, TimeUnit.SECONDS)
+        .readTimeout(180, TimeUnit.SECONDS)
+        .callTimeout(180, TimeUnit.SECONDS)
+        .writeTimeout(180, TimeUnit.SECONDS)
```

Comment:
These timeouts should be configurable

Reply:
A note about this: they're the highest timeout possible. I've increased to 5 minutes.